File: joseffe-barroso/Python/aula04/while/exercise46.py
quantity = int(input("Informe a quantidade: "))
while (quantity <= 0 or quantity > 20): 
  quantity = int(input("O valor da sequência deve ser positivo e menor que vinte. Informe novamente: "))

sumNumbers = 0
higherNumber = 0
smallestNumber = 0
avaragePositiveValues = 0
avarageNegativeValues = 0
userAnswer = 'NAO'

index = 0
while (index < quantity):
  number = int(input("Digite um número: "))

  if (number <= 0):
    avarageNegativeValues = avarageNegativeValues + 1
  elif (number > 0):
    avaragePositiveValues = avaragePositiveValues + 1

  if (index == 0):
    higherNumber = number
    smallestNumber = number
  elif (number > higherNumber):
    higherNumber = number
  elif (number < smallestNumber):
    smallestNumber = number

  sumNumbers = sumNumbers + number

  avarage = sumNumbers / quantity
   
  index = index + 1

  if (index == quantity):
    print(f"Maior valor informado: {higherNumber}")
    print(f"Menor valor informado: {smallestNumber}")
    print(f"Soma dos valores informado: {sumNumbers}")
    print(f"Média dos valores informados: {avarage}")
    print(f"Porcentagem dos valores positivos: {(avaragePositiveValues * 100) / quantity}%")
    print(f"Porcentagem dos valores negativos: {(avarageNegativeValues * 100) / quantity}%")
    
    userAnswer = input("\nDeseja executar o programa novamente? 'SIM' ou 'NAO': ")
    while (userAnswer != 'SIM' and userAnswer != 'NAO'): 
      userAnswer = input("Deseja executar o programa novamente? 'SIM' ou 'NAO' ")
    if (userAnswer == 'SIM'): 
      sumNumbers = 0
      higherNumber = 0
      smallestNumber = 0
      avaragePositiveValues = 0
      avarageNegativeValues = 0
      quantity = int(input("Informe a quantidade: "))
      while (quantity <= 0 or quantity > 20): 
        quantity = int(input("O valor da sequência deve ser positivo e menor que vinte. Informe novamente: "))
      index = 0
    

File: joseffe-barroso/Java/src/aula04/While/Exercise46.java
package While;

import java.util.Scanner;

public class Exercise46 {

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);

		String novocalculo;
		double n, i, soma, xneg, xpos, maior, menor, xnul, x, media, porcneg, porcpos, porcnul;

		novocalculo = "S";

		while (novocalculo.toUpperCase().equals("S")) {

			System.out
					.println("Digite a quantidade de números a ser comparada (deve ser maior que 0 e menor que 20): ");
			n = scanner.nextDouble();

			while ((n < 1) || (n >= 20)) {
				System.out.println("Número inválido. O número digitado deve estar entre 1 e 19.");
				System.out.println(
						"Digite a quantidade de números a ser comparada (deve ser maior que 0 e menor que 20): ");
				n = scanner.nextDouble();
			}
			i = 1;
			xpos = 0;
			xneg = 0;
			xnul = 0;
			soma = 0;
			maior = 0;
			menor = 0;

			while (i <= n) {
				System.out.println("Digite o " + (int) i + "º número: ");
				x = scanner.nextDouble();

				if (i == 1) {
					maior = x;
					menor = x;
					if (x > 0) {
						xpos = xpos + 1;
					} else if (x < 0) {
						xneg = xneg + 1;
					} else {
						xnul = xnul + 1;
					}

				} else if (x > maior) {
					maior = x;
					if (x > 0) {
						xpos = xpos + 1;
					} else if (x < 0) {
						xneg = xneg + 1;
					} else {
						xnul = xnul + 1;
					}
				} else if (x < menor) {
					menor = x;
					if (x > 0) {
						xpos = xpos + 1;
					} else if (x < 0) {
						xneg = xneg + 1;
					} else {
						xnul = xnul + 1;
					}
				} else {
					if (x > 0) {
						xpos = xpos + 1;
					} else if (x < 0) {
						xneg = xneg + 1;
					} else {
						xnul = xnul + 1;
					}
				}
				soma = soma + x;
				i++;
			}
			media = soma / n;
			porcneg = 100 * xneg / n;
			porcpos = 100 * xpos / n;
			porcnul = 100 * xnul / n;

			System.out.println("O maior valor é: " + maior);
			System.out.println("O menor valor é: " + menor);
			System.out.println("A soma dos valores é: " + soma);
			System.out.println("A média dos valores positivos é: " + media);
			System.out.println("A porcentagem dos valores positivos é: " + porcpos);
			System.out.println("A porcentagem dos valores negativos é: " + porcneg);
			System.out.println("A porcentagem dos valores nulos (0) é: " + porcnul);

			System.out.println("Deseja efetuar novo cálculo? (S para SIM e N para NÃO) ");
			novocalculo = scanner.next();

			while (!novocalculo.toUpperCase().equals("S") && (!novocalculo.toUpperCase().equals("N"))) {
				System.out.println("Resposta inválida.");
				System.out.println("Deseja efetuar novo cálculo? (S para SIM e N para NÃO) ");
				novocalculo = scanner.next();
			}
		}
		System.out.println("Programa Encerrado.");

		scanner.close();
	}

}
